addons/sale/models/sale_order_line.py

```diff
@@ -1293,4 +1292,5 @@ def has_valued_move_ids(self):
         return self.move_ids
 
     def _sellable_lines_domain(self):
-        return [('is_downpayment', '=', False)]
+        discount_products_ids = self.env.companies.sale_discount_product_id.ids or []
+        return ['&', ('is_downpayment', '=', False), ('product_id', 'not in', discount_products_ids)]
```

Comment:
We could avoid adding `('product_id', 'not in', [])` if we are sure the right part of the condition in that tuple is an empty list. 🙂 

Something like:

```suggestion
        discount_product_ids = self.env.companies.sale_discount_product_id.ids
        domain = [('is_downpayment', '=', False)]
        if discount_product_ids:
            domain = expression.AND([
                domain,
                [('product_id', 'not in', discount_products_ids)],
            ])
        return domain
```

What do you think? 🙂